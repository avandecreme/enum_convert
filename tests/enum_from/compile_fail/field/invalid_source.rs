use enum_convert::EnumFrom;

enum Source {
    Data { x: i32 },
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    #[enum_from(Source::Data)]
    Stuff {
        #[enum_from(Source::Data.nonexistent)] // Invalid field name
        x: i64,
    },
}

fn main() {}
