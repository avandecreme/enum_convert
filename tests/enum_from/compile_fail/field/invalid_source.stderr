error[E0026]: variant `Source::Data` does not have a field named `nonexistent`
  --> tests/enum_from/compile_fail/field/invalid_source.rs:12:29
   |
12 |         #[enum_from(Source::nonexistent)] // Invalid field name
   |                             ^^^^^^^^^^^ variant `Source::Data` does not have this field

error: pattern requires `..` due to inaccessible fields
 --> tests/enum_from/compile_fail/field/invalid_source.rs:7:10
  |
7 | #[derive(EnumFrom)]
  |          ^^^^^^^^
  |
  = note: this error originates in the derive macro `EnumFrom` (in Nightly builds, run with -Z macro-backtrace for more info)
help: ignore the inaccessible and unused fields
  |
12|         #[enum_from(Source::nonexistent, ..)] // Invalid field name
  |                                        ++++
